### PR TITLE
✨Introduce fallback logic for remove signing

### PR DESCRIPTION
**Pull request by calebcordry**
Hopefully last big PR :) 

Part of #27189

### CODE REVIEW

**Comment by calebcordry** [comment]
looking into test failures.

**Comment by calebcordry** [comment]
@dvoytenko Could you please take a look for `src/*` whenever you can?

File: extensions/amp-a4a/0.1/amp-a4a.js [hunk 1]
```diff
@@ -1614,59 +1648,58 @@ export class AmpA4A extends AMP.BaseElement {
...
+      this.getIframeTitle(),
+      height,
+      width
+    );
+    this.applyFillContent(this.iframe);
 
-      // TODO(ccordry): FIE does not handle extension versioning, but we have
-      // it accessible here.
-      const extensionIds = extensions.map((extension) => extension.extensionId);
+    // TODO(ccordry): FIE does not handle extension versioning, but we have
```
Comment (dvoytenko) [inline extensions/amp-a4a/0.1/amp-a4a.js:1671]: Not a subject to this pull request, but I don't think there's any reason left to not support versions. They just need to be passed through - that's it. Could you pls schedule a follow up on this?

File: src/utils/stream-response.js [hunk 2]
```diff
@@ -14,21 +14,30 @@
...
+import {Deferred} from './promise';
+
 /**
  * Decodes readable stream from response and writes to given writeable stream.
  * This function should be replaced with transform stream when well supported.
  * @param {!Window} win
  * @param {!Response} response
  * @param {!./detached-dom-stream.DetachedDomStream} writer
- * @return {!Promise}
+ * @return {!Promise<boolean>} Indicates if the response is empty.
```
Comment (dvoytenko) [inline src/utils/stream-response.js:25]: Can you clarify the doc here? Does this yield `true` when stream is empty or not empty? TBH, neither sounds very good from the API standpoint. But let's see if we can at least clarify it in docs.

File: extensions/amp-a4a/0.1/amp-a4a.js (hunk 1 above)
Comment (calebcordry) [reply extensions/amp-a4a/0.1/amp-a4a.js:1671]: Sure, created #29604

File: src/utils/stream-response.js (hunk 2 above)
Comment (calebcordry) [reply src/utils/stream-response.js:25]: It returns `false` if empty. We need some signal if the response has content or not, but if you have concerns over the API I should probably change it?

Comment (calebcordry) [reply src/utils/stream-response.js]: Bugged Dima offline :) we will leave for now, and can change if we think of a better API in future. Edited JSDoc to try to help clarify the behavior.

**Pull request merged**